build.py

```diff
@@ -341,6 +344,91 @@ def step_end():
         subprocess.run((cargo_args + ["--", "--qsharp-json", project]), check=True, text=True, cwd=root_dir)
     step_end()
 
+if build_notebooks:
+    step_start("Running notebook samples")
+    # Find all notebooks in the samples directory. Skip the basic sample and the azure submission sample, since those won't run
+    # nicely in automation.
+    notebook_files = [os.path.join(dp, f) for dp, _, filenames in os.walk(samples_src) for f in filenames if f.endswith(".ipynb")
+                      and not (f.startswith("sample.") or f.startswith("azure_submission."))]
```

Comment:
Do we trust ourselves not to check in sample notebooks with arbitrary web requests, pip installs, and other stuff that might make automation flaky? Should notebooks be explicitly included instead?

Reply:
That's a interesting question... my thinking with going with an exclude list instead is that it makes adding a new sample easier, and any misbehaving samples will (best case) get caught when the PR pipeline runs before they merge or (worst case) show as flakiness later that forces us to stop and take a look at why that notebook is flaky. Then we will be incentivised to keep our notebook samples well-contained and cleanly running, instead of defaulting to not testing them and having to remember to add good ones to the list here.

Reply:
Fair enough - we can always change later.